Floofbot/Modules/Administration.cs

```diff
@@ -966,5 +912,142 @@ private async Task SetUserNoteForgivenStatus(UserNote un, bool status, ulong for
             un.ForgivenBy = forgivenBy;
             await _floofDB.SaveChangesAsync();
         }
+        private async Task<Embed> GetWarnings(ulong uid, bool isOwnLog, ISocketMessageChannel channel)
+        {
+            try {
+                IQueryable<Warning> formalWarnings = null;
+                IQueryable<UserNote> userNotes = null;
+                SocketUser badUser = Context.Client.GetUser(uid);
+
+                if (isOwnLog)
+                {
+                    formalWarnings = _floofDB.Warnings.AsQueryable()
+                        .Where(u => u.UserId == uid && u.GuildId == Context.Guild.Id && u.Forgiven == false)
+                        .OrderByDescending(x => x.DateAdded).Take(10);
+                }
+                else // user notes are for mod view only
+                {
+
+                    formalWarnings = _floofDB.Warnings.AsQueryable()
+                        .Where(u => u.UserId == uid && u.GuildId == Context.Guild.Id)
+                        .OrderByDescending(x => x.DateAdded).Take(10);
+
+                    userNotes = _floofDB.UserNotes.AsQueryable()
+                        .Where(u => u.UserId == uid && u.GuildId == Context.Guild.Id)
+                        .OrderByDescending(x => x.DateAdded).Take(10);
+                }
+
+                if (!isOwnLog) // mod viewing someones history
+                {
+                    if (badUser == null) // client cant get user - no mutual servers?
+                    {
+                        if (formalWarnings.Count() == 0 && userNotes.Count() == 0)
+                        {
+                            await channel.SendMessageAsync($"{uid} is a good noodle. They have no warnings or user notes!");
+                            return null;
+                        }
+                    }
+                    else
+                    {
+                        if (formalWarnings.Count() == 0 && userNotes.Count() == 0)
+                        {
+                            await channel.SendMessageAsync($"{badUser.Username}#{badUser.Discriminator} is a good noodle. They have no warnings or user notes!");
+                            return null;
+                        }
+                    }
+                }
+                else // own users history
+                {
+                    if (formalWarnings.Count() == 0)
+                    {
+                        await channel.SendMessageAsync($"You are a good noodle. You have no warnings!");
```

Comment:
There is some privacy leakage here. Any user who uses `warnlog` and does _not_ get this response in the channel they sent the message in will be a dead giveaway they have at least one warning. As an optional refactor, the bot should never send a response to the `warnlog` command in the same channel and instead tell the user in a DM that they have warnings or are a good noodle.

This can probably be achieved by modifying this line and moving it to the logic on [R435](https://github.com/bealsbe/Floofbot/pull/90/files#diff-2f88d76a18ee817ca5439e76a15ab026R435)